commit 9098bc1dfb64e38570aec4c4d4fff0ebea29fb75
Author: Colin Guthrie <colin.guthrie@tribalogic.net>
Date:   Thu Mar 19 15:41:06 2009 +0000

    Ensure Zend_Controller_Router_Route_Regexp properly honours the $reset argument.

diff --git a/Zend/Controller/Router/Route/Regex.php b/Zend/Controller/Router/Route/Regex.php
index 15daf58..8c4a014 100644
--- a/Zend/Controller/Router/Route/Regex.php
+++ b/Zend/Controller/Router/Route/Regex.php
@@ -151,7 +151,10 @@ class Zend_Controller_Router_Route_Regex extends Zend_Controller_Router_Route_Ab
         }
 
         $defaultValuesMapped  = $this->_getMappedValues($this->_defaults, true, false);
-        $matchedValuesMapped  = $this->_getMappedValues($this->_values, true, false);
+        if ($reset)
+            $matchedValuesMapped = array();
+        else
+            $matchedValuesMapped  = $this->_getMappedValues($this->_values, true, false);
         $dataValuesMapped     = $this->_getMappedValues($data, true, false);
 
         // handle resets, if so requested (By null value) to do so
